### DISCUSSION TITLE
Are useFieldArray's IDs guaranteed to be unique?

### DISCUSSION THREAD

**Discussion by wondering639** [react-hook-form/react-hook-form#5161] [Q&A] [answered]
The UUIDs that `useFieldArray `provides by default in the `id` property, are they guaranteed to be unique? Meaning after a new UUID is generated is there a check that's it's really unique and not by chance a UUID that was generated before? (I know this is very unlikely, though not impossible by chance?)

**Comment by thanh-nguyen-95** [comment, marked as answer]
There's always a chance of duplication though it's very low. If you're planning on using it for your data, I recommend you don't.

**Reply by wondering639** [reply]
thanks for your reply. I don't want to use them in the database, just for the React key property. I guess there it's the same issue: unlikely to the equal, but not guaranteed, correct?

**Reply by thanh-nguyen-95** [reply]
That is correct

**Reply by wondering639** [reply]
Thank you. Do you have a reference to the source code where the id is generated?
Edit: guess I found it:
https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/generateId.ts
called by
https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/mapId.ts
called by
https://github.com/react-hook-form/react-hook-form/blob/master/src/useFieldArray.ts
